### DISCUSSION TITLE
i want to know what i did wrong?

### DISCUSSION THREAD

**Discussion by hacker101red** [pokepetter/ursina#261] [Q&A]
``from` `ursina` `import` `*`
`from` `random` import randint 
import math
from ursina.prefabs.platformer_controller_2d import PlatformerController2d

app = Ursina()
window.title = 'My Game'  
window.iconbitmap="bb.ico"             
window.exit_button.visible = True
window.minimize_button=False
window.background='background'
Sky(texture="sky.jpg")
entities=[]

player=PlatformerController2d(scale_y=1, x=-24,jump_height=5, collider="box", y=10)
ground = Entity(model='quad', y=-2, scale_x=50, collider="box", scale_y=10, texture = "ground")
camera.add_script(SmoothFollow(target=player, offset=[0,1,-30]))
ground2 = Entity(model='quad', y=-2, scale_x=50, collider="box", scale_y=10, texture = "ground")
wall=Entity(model="quad", x=-25, scale_y=50, collider="box", texture="wall")
spikes=Entity(model="spike", x=27, collider="box", scale=0.05, y=-5)
wall2=Entity(model="quad", x=-26,scale_y=50, collider="box", texture="wall")
ground2 = Entity(model='quad', y=-2, scale_x=50, collider="box", scale_y=10, texture = "ground", x=50)

def update():

     
     red=randint(0, 255) 
     blue=randint(0, 255) 
     green=randint(0, 255)
     player.color=rgb(red, green, blue)
   
     player.x += held_keys['w'] * .1
     player.x -= held_keys['s'] * .1
     if held_keys["x"]:
      camera.rotation_x=camera.rotation_x+time.dt*100
     if held_keys["y"]:  
      camera.rotation_y=camera.rotation_y+time.dt*100
     hiti=spikes.intersects()     
     if hiti.hit:
          txt=Text(text="game over", scale=5, position=(0,0,0))
          player_y=-50
     if hiti.hit:
          player_y=-50
EditorCamera()
     

     

app.run()``

my code

**Comment by pokepetter** [comment]
What's the issue?

**Comment by PixelTomato** [comment]
Could you please tell us what is wrong? We do not have access to your spike model so if the issue is related to the "game over" text instantly appearing, we may not be able to help until we know what the collision box looks like for that model.

**Reply by hacker101red** [reply]
i wrote it in my previous question

**Reply by hacker101red** [reply]
and the collision is if  the player hit the spikes it will show game over kinda like mario

**Reply by hacker101red** [reply]
spikes link:https://drive.google.com/file/d/1LInA4vsbFdCZZ7XHk-m-fbC2zmXXwTdu/view?usp=sharing

**Comment by PixelTomato** [comment]
To fix the instantaneous "Game Over" screen, you can change line 37 to this:
```py
hiti=spikes.intersects(player)
```
Passing in the player object causes `.hit()` to only return true when intersecting with the player.

The issue that arises here is that the `PlatformerController2d()` class already has functions that manage collisions before your code does which means that the player will likely not have the chance to register collision events before it is moved outside of any colliders it may enter.

**Reply by hacker101red** [reply]
thx i will try
